src/aof.c

```diff
@@ -653,15 +1051,10 @@ void feedAppendOnlyFile(int dictid, robj **argv, int argc) {
     /* Append to the AOF buffer. This will be flushed on disk just before
      * of re-entering the event loop, so before the client will get a
      * positive reply about the operation performed. */
-    if (server.aof_state == AOF_ON)
+    if (server.aof_state == AOF_ON ||
+        (server.aof_state == AOF_WAIT_REWRITE && server.child_pid != -1)) {
```

Comment:
styling. `if` with multiple lines should move the `{` to the next line (less confusing indentation).
```suggestion
    if (server.aof_state == AOF_ON ||
        (server.aof_state == AOF_WAIT_REWRITE && server.child_pid != -1))
    {
```